pkg/autoscaler/autoscaler.go

```diff
@@ -279,145 +281,218 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 			Expect(client.Create(ctx, workload)).Should(Succeed())
 
-			for _, machineSet := range transientMachineSets {
-				By(fmt.Sprintf("Waiting for MachineSet %s replicas to scale out", machineSet.GetName()))
-			}
-			Eventually(func() (bool, error) {
-				for _, machineSet := range transientMachineSets {
-					current, err := framework.GetMachineSet(client, machineSet.GetName())
-					if err != nil {
-						return false, err
-					}
-					if *current.Spec.Replicas != expectedReplicas {
-						return false, nil
-					}
+			By(fmt.Sprintf("Waiting for MachineSet %s replicas to scale out", machineSet.GetName()))
+			Eventually(func() (int32, error) {
+				current, err := framework.GetMachineSet(client, machineSet.GetName())
+				if err != nil {
+					return 0, err
 				}
-				return true, nil
-			}, framework.WaitMedium, pollingInterval).Should(BeTrue())
+				return *current.Spec.Replicas, nil
+			}, framework.WaitMedium, pollingInterval).Should(BeEquivalentTo(expectedReplicas))
 
-			By("Waiting for all Machines in MachineSets to enter Running phase")
-			framework.WaitForMachineSet(client, transientMachineSets[0].GetName())
-			framework.WaitForMachineSet(client, transientMachineSets[1].GetName())
+			By("Waiting for all Machines in the MachineSet to enter Running phase")
+			framework.WaitForMachineSet(client, machineSet.GetName())
 
 			By("Deleting the workload")
 			Expect(deleteObject(workload.Name, cleanupObjects[workload.Name])).Should(Succeed())
 			delete(cleanupObjects, workload.Name)
 
-			for _, machineSet := range transientMachineSets {
-				By(fmt.Sprintf("Waiting for MachineSet %s replicas to scale in", machineSet.GetName()))
-			}
+			By(fmt.Sprintf("Waiting for MachineSet %s replicas to scale in", machineSet.GetName()))
 			expectedLength := 1
-			Eventually(func() (bool, error) {
-				for _, machineSet := range transientMachineSets {
-					machines, err := framework.GetMachinesFromMachineSet(client, machineSet)
+			var machines []*machinev1.Machine
+			Eventually(func() (int, error) {
+				machines, err = framework.GetMachinesFromMachineSet(client, machineSet)
+				if err != nil {
+					return 0, err
+				}
+				return len(machines), nil
+			}, framework.WaitLong, pollingInterval).Should(BeEquivalentTo(expectedLength))
+
+			for _, machine := range machines {
+				By(fmt.Sprintf("Checking Machine %s for %s annotation", machine.Name, machineDeleteAnnotationKey))
+				Eventually(func() (bool, error) {
+					m, err := framework.GetMachine(client, machine.Name)
 					if err != nil {
 						return false, err
 					}
-					if len(machines) != expectedLength {
+					if m.ObjectMeta.Annotations == nil {
+						return true, nil
+					}
+					if _, exists := m.ObjectMeta.Annotations[machineDeleteAnnotationKey]; exists {
 						return false, nil
 					}
-				}
-				return true, nil
-			}, framework.WaitLong, pollingInterval).Should(BeTrue())
+					return true, nil
+				}, framework.WaitMedium, pollingInterval).Should(BeTrue())
```

Comment:
just curious why this was switched to a `WaitMedium` instead of `WaitLong`

Reply:
It was not. This is just confusing diff. There are three eventually blocks next to each other. The first uses WaitLong and the other two use WaitMedium.

Reply:
ack, thanks!